cmd/lotus-shed/indexes.go

```diff
@@ -196,6 +202,11 @@ var backfillEventsCmd = &cli.Command{
 			var eventsAffected int64
 			var entriesAffected int64
 
+			tsKeyCid, err := currTs.Key().Cid()
```

Comment:
Hard to see this while reviewing on my phone but I think this is only being set once here but we need to do it each time we update it so we're inserting the right one. `currTs` should follow the epoch as we move through the history and its CID should be the thing we insert.

Reply:
nope, my phone deceived me, this is a good optimisation

Reply:
If you see [processHeight](https://github.com/akaladarshi/lotus/blob/76d29d44b3a93f54627e945ebabdb2dec49cc029/cmd/lotus-shed/indexes.go#L356) is called first then we are updating the [currTs](https://github.com/akaladarshi/lotus/blob/76d29d44b3a93f54627e945ebabdb2dec49cc029/cmd/lotus-shed/indexes.go#L363), and It's not an async call so `currTs` won't be updated till `processHeight` is finished.

Let me know if I am missing something.